src/lifetimes.rs

```diff
@@ -61,6 +64,7 @@ fn check_fn_inner(cx: &LateContext, decl: &FnDecl, slf: Option<&ExplicitSelf>,
         span_lint(cx, NEEDLESS_LIFETIMES, span,
                   "explicit lifetimes given in parameter types where they could be elided");
     }
+    report_extra_lifetimes(cx, decl, &generics.lifetimes);
```

Comment:
I'm not sure, but won't this report methods in trait impls where the lifetime is specified on the trait definition? Would it be legal to remove it while leaving it on the trait def? I'm on mobile, so I can't check right now. I'd really like if fixing one lint warning (as opposed to all of them) doesn't introduce compilation errors.